docs/src/literate-tutorials/dg_heat_equation.jl

```diff
@@ -0,0 +1,411 @@
+∂Ωₙ = union(
+    getfaceset(grid, "top"),
+    getfaceset(grid, "bottom"),
+);
+
+# Now we are set up to define our constraint. We specify which field
+# the condition is for, and our face sets. The last
+# argument is a function of the form $f(\textbf{x})$ or $f(\textbf{x}, t)$,
+# where $\textbf{x}$ is the spatial coordinate and
+# $t$ the current time, and returns the prescribed value.
+add!(ch, Dirichlet(:u, getfaceset(grid, "right"), (x, t) -> 1.0));
+add!(ch, Dirichlet(:u, getfaceset(grid, "left"), (x, t) -> -1.0));
+
+# Finally we also need to `close!` our constraint handler. When we call `close!`
+# the dofs corresponding to our constraints are calculated and stored
+# in our `ch` object.
+close!(ch)
+
+# Note that if one or more of the constraints are time dependent we would use
+# [`update!`](@ref) to recompute prescribed values in each new timestep.
+
+# ### Assembling the linear system
+#
+# Now we have all the pieces needed to assemble the linear system, $K u = f$.
+# Assembling of the global system is done by looping over all the elements in order to
+# compute the element contributions ``K_e`` and ``f_e``, all the interfaces
+# to compute their contributions ``K_i``, and all the Neumann boundary faces to compute their
+# contributions ``f_e`` which are then assembled to the
+# appropriate place in the global ``K`` and ``f``.
+#
+# #### Element assembly
+# We define the functions
+# * `assemble_element!` to compute the contribution of volume integrals over an element.
+# * `assemble_interface!` to compute the contribution of surface integrals over an interface.
+# * `assemble_boundary!` to compute the contribution of surface integrals over a boundary face.
+#
+# The function takes pre-allocated `ke`, `ki`, and `fe` (they are allocated once and
+# then reused for all elements) so we first need to make sure that they are all zeroes at
+# the start of the function by using `fill!`. Then we loop over all the quadrature points,
+# and for each quadrature point we loop over all the (local) shape functions. The *values
+# objects provide values needed to calculate contributions as follows:
+# * `cellvalues`:
+#     * test function value ($\delta u$)
+#     * test and trial functions gradient ($\nabla u$ and $\nabla \delta u$)
+# * `facevalues`:
+#     * test function value ($δu$)
+# * `interfacevalues`:
+#     * jump in test and trial functions values
+#     * average of test and trial functions gradients
+#
+# !!! note "Notation"
+#     Comparing with the brief finite element introduction in [Introduction to FEM](@ref),
+#     the variables `δu`, `∇δu` and `∇u` are actually $\phi_i(\textbf{x}_q)$, $\nabla
+#     \phi_i(\textbf{x}_q)$ and $\nabla \phi_j(\textbf{x}_q)$, i.e. the evaluation of the
+#     trial and test functions in the quadrature point ``\textbf{x}_q``. However, to
+#     underline the strong parallel between the weak form and the implementation, this
+#     example uses the symbols appearing in the weak form.
```

Comment:
Is this needed here? 
It follows the basic heat equation notation anyways. 
But on that note, perhaps even add a remark in the beginning that readers are expected to be familiar with the heat equation tutorial before reading this one?